pycontrails/models/cocipgrid/cocip_grid.py

```diff
@@ -342,6 +389,105 @@ def _bundle_results(
     # Common Methods & Properties
     # ---------------------------
 
+    @property
+    def source_time(self) -> npt.NDArray[np.datetime64]:
+        """Return the time array of the :attr:`source` data."""
+        try:
+            source = self.source
+        except AttributeError as exc:
+            msg = "No source set"
+            raise AttributeError(msg) from exc
+
+        if isinstance(source, GeoVectorDataset):
+            return source["time"]
+        if isinstance(source, MetDataset):
+            return source.variables["time"].values
+
+        msg = f"Cannot calculate timesteps for {source}"
+        raise TypeError(msg)
+
+    def _set_timesteps(self) -> None:
+        """Set the :attr:`timesteps` based on the ``source`` time range."""
+        source_time = self.source_time
+        tmin = source_time.min()
+        tmax = source_time.max()
+
+        tmin = pd.to_datetime(tmin)
+        tmax = pd.to_datetime(tmax)
+        dt = pd.to_timedelta(self.params["dt_integration"])
+
+        t_start = tmin.ceil(dt)
```

Comment:
This is an interesting thought. Other folks using pycontrails have requested this sort of feature (it's especially useful for comparing CoCiP outputs to GOES imagery occurring with a fixed offset). What we have here is somewhat arbitrary ... I'm not sure what would make the most sense. I think someone could override or monkey patch `CocipGrid._set_timesteps` to customize the choice here. It might be cleaner to include an model parameter `timesteps_override` that could be a callable or numpy array.